Taskfile.yml

```diff
@@ -0,0 +1,168 @@
+# https://taskfile.dev
+# To run on a system without task installed, `./scripts/run_task.sh` will execute it with `go run`.
+# If in the nix dev shell, `task` is available.
+
+version: '3'
+
+tasks:
+  default: ./scripts/run_task.sh --list
+
+  actionlint:
+    desc: Run the actionlint.sh
+    cmd: ./scripts/actionlint.sh # tests.yml
+
+  build:
+    desc: Run the build.sh
+    cmd: ./scripts/build.sh # tests.yml
+
+  build-antithesis-images:
+    desc: Build the antithesis images for the avalanchego test setup
+    cmd: bash -x ./scripts/build_antithesis_images.sh # publish_antithesis_images.yml
+
+  build-bench-precompiles:
+    desc: Build the benchmark precompiles
+    cmd: ./scripts/build_bench_precompiles.sh # bench.yml
+
+  build-docker-image:
+    desc: Build the docker image
+    cmd: ./scripts/build_docker_image.sh # publish_docker.yml
+
+  build-test:
+    desc: Run the build_test.sh
+    cmd: ./scripts/build_test.sh # tests.yml
+
+  check-clean-branch:
+    desc: Checks that the git working tree is clean
+    cmd: .github/workflows/check-clean-branch.sh
+
+  check-generate-codec:
+    desc: Checks that generated codec files are up-to-date (requires a clean git working tree)
+    cmds:
+      - task: generate-codec
+      - task: check-clean-branch
+
+  check-generate-mocks:
+    desc: Checks that generated mocks are up-to-date (requires a clean git working tree)
+    cmds:
+      - task: generate-mocks
+      - task: check-clean-branch
+
+  check-go-mod-tidy:
+    desc: Checks that go.mod and go.sum are up-to-date (requires a clean git working tree)
+    cmds:
+      - cmd: go mod tidy
+      - task: check-clean-branch
+
+  coverage:
+    desc: Run the coverage.sh
+    cmd: ./scripts/coverage.sh # tests.yml
+
+  generate-codec:
+    desc: Generates fjl/gencodec files
+    cmds:
+      - cmd: grep -lr -E '^// Code generated by github\.com\/fjl\/gencodec\. DO NOT EDIT\.$' . | xargs -r rm
+      - cmd: go generate -run "github.com/fjl/gencodec" ./...
+
+  generate-mocks:
+    desc: Generates testing mocks
+    cmds:
+      - cmd: grep -lr -E '^// Code generated by MockGen\. DO NOT EDIT\.$' . | xargs -r rm
+      - cmd: go generate -run "go.uber.org/mock/mockgen" ./...
+
+  install-avalanchego-release:
+    desc: Install AvalancheGo release for testing
+    env:
+      BASEDIR: '{{.BASEDIR | default "/tmp/e2e-test"}}'
```

Comment:
This is actually required here to tell the antithesis tests where to find the binary - moving the env the the task specifically (see line 112) makes sense but still results in the path being unable to be found, as tasks [do not inherit environment variables from the tasks that call them](https://github.com/go-task/task/issues/1277). I am okay leaving it as it is, but curious as to your opinion.

Reply:
Ah I'll set it globally!